keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -478,46 +486,49 @@ Because Windows privileged containers will work much differently than Linux priv
-- Unix domain sockets mounts support will be added before `HostProcess` containers graduate to `stable`. For `alpha` and `beta` Unix domain sockets can be accessed via their paths on the host like named pipes.
-  - The Windows APIs needed to support mounting unix domain socket mounts in `hostProcess` containers was introduced in Windows Server Ver, 2004. Microsoft is planning on backporting these APIs to Windows Server 2019 (min support Windows Server OS) to provide a consistent user experience.
-- Mounting directories from the host OS into `hostProcess` containers will work just like with normal containers - kubelet will explicitly block this scenario.
-- All other volume types supported for normal containers on Windows will work with `hostProcess` containers.
+- Window's bind-filter driver will be used to create a view that merges the host's OS filesystem with container-local volumes.
+When `hostProcess` containers are started a new volume will be created which contains the contents of the contaner image.
+This volume will be mounted to `c:\hpc`. The default working directory for `hostProcess` containers will also be set to `c:\hpc`.
+- Volume mounts (includinge service account tokens) will be supported for `hostProcess` containers and can be accessed just the same way as regular Windows Server containers.
+  - Named Pipe mounts will **not** be supported.
+    Instead named pipes should be accessed via their path on the host (\\\\.\\pipe\\*).
+    The following error will be returned if `hostProcess` containers attempt to use name pipe mounts -
+    https://github.com/microsoft/hcsshim/blob/358f05d43d423310a265d006700ee81eb85725ed/internal/jobcontainers/mounts.go#L40.
+  - Unix domain sockets mounts also not not be supported for `hostProces` containers.
+    Unix domain sockets can be accessed via their paths on the host like named pipes.
+  - Mounting directories from the host OS into `hostProcess` containers will work just like with normal containers but this is not recommend.
+    Instead workloads should access the host OS's file-system as if it was not being run in a container.
+  - All other volume types supported for normal containers on Windows will work with `hostProcess` containers.
+- `HostProcess` Containers will have full access to the host file-system (unless restricted by filed-based ACLs and the run_as_username used to start the container).
+- There will be no `chroot` equivalent.
+
+- Note: Behavior of volume mounts will differ between the alpha/beta (old) implementation of this feature and the stable (new) implementation.
+Designing/testing/validation of an acceptable solution for handling volume mounts w.r.t. `hostProcess` containers was a primary reason for keeping the featuer in `beta`. (Previous behavior volume mount behavior can be viewed [here](https://github.com/kubernetes/enhancements/blob/cc4052fa88508fcbb20b978b8c181d06acf734b1/keps/sig-windows/1981-windows-privileged-container-support/README.md#container-mounts))
```

Comment:
Has the new behavior been implemented yet? Should we consider keeping the feature in beta to allow more time for testing and refining the new behavior?

Reply:
This functionality (minus the annotation used to request either old or new behavior) has already been implemented.

So far all of the feedback we've received around this feature has been positive and everyone has asked when it will graduate out of beta so other work can take decencies on host-process containers (like deployment kube-proxy/CNI solutions via daemonSets as part of `kubeadm` workflows)
Also most of the work for needed for changing the volume mount behavior was done in components below the kubelet (containerd, hcsshim, and Windows OS) and also ship independently from K8s.

Because of this I don't have concerns moving out of beta in 1.25.
Curious what other folks (@jsturtevant, @jayunit100 , @claudiubelu) think.

Reply:
We have plans to switch over all the tests for sig-windows and Cluster Api Provider Azure to use the new behavior (where this is used for CNI's and kube-proxy, CSI proxy and a few other components).  I don't anticipate any issues from a testing perspective and will have a good signal by the time 1.25 finalizes.

Since this new behavior is _expected_ behavior for a User of HPC (instead of having to play with strange `CONTAINER_SANDBOX_MOUNT_POINT`  paths)  I feel good about it from a refinement perspective.